src/inventory.go

```diff
@@ -76,7 +78,13 @@ func populateInventory(reader *bufio.Reader, i *inventory.Inventory) error {
 		case '#':
 			// ignore comments
 			for r != '\n' {
-				r, _, _ = reader.ReadRune()
+				r, _, err = reader.ReadRune()
+				if err != nil {
+					if errors.Is(err, io.EOF) {
+						return nil
+					}
+					return fmt.Errorf("ignoring comment at line %d: %w", lineNo, err)
```

Comment:
the message is a bit misleading, we are not ignoring the line since we are stopping the execution because of it.

By the way few lines above the error management is different:
```
			if err != nil {
				continue // Break to outer loop so we can handle errors/EOF
			}
```
I like yours more since we are checking the error in place instead of trying to read again the line for some reason, could we have the same?

Reply:
Updated here c5ed5426115bb8be7957a1cc1b79233f80efdc4c